### PR TITLE
Migration History: audit DDL from the SQL runners (closes #120)

### LINKED ISSUE #120: Migration History: track DDL from all sources (SQL runner, MCP, direct), not just the platform DDL hook
## Problem

Today only DDL that flows through the platform's typed DDL handlers (\`HandleCreateTable\`, \`HandleAddColumn\`, etc.) reaches \`schema_changes\` via \`logSchemaChange\`. Every other path silently bypasses the audit:

- **Console SQL runner** (\`/platform/projects/{id}/data/sql\` → \`query.HandlePlatformSQL\`) — runs arbitrary SQL, no logging hook.
- **Console SQL runner transaction** (\`/data/sql/transaction\`) — same.
- **MCP \`runSQL\` / \`runSQLTransaction\`** — same.
- **SDK \`/v1/db/sql\`** — same.
- **Anything done via direct DB access** — out of platform control entirely.

Surfaced by today's #119 postmortem: an \`ALTER TABLE app_users ADD COLUMN stable_device_id\` issued via MCP \`runSQL\` does not appear in Migration History at all. PR #119 fixed the *backfill* mis-stamping, but the underlying gap — that the migration history is a partial ledger — remains.

## Acceptance

**\"ALL migrations, regardless of source, should be in the list.\"** — every CREATE / ALTER / DROP / RENAME / INDEX-ish change on tenant schemas, regardless of whether it was issued via the console SQL editor, MCP, the SDK's \`runSQL\`, or anything else, should show up in Migration History with an accurate timestamp.

## Approach (suggested)

**A. Postgres event trigger** — the right answer.

Event triggers fire on \`ddl_command_end\` regardless of how the command reached Postgres. A trigger function in \`public.*\` (owned by \`eurobase_migrator\`, deployed via a migration) walks \`pg_event_trigger_ddl_commands()\`, derives the project_id from \`schema_name\` → \`projects\` lookup, and inserts the appropriate \`schema_changes\` row. One trigger covers every code path forever.

Pros:
- Catches direct DB access too — defense in depth for the audit log.
- No code change in the SQL runner / MCP server / SDK; the audit is at the DB layer.
- Survives future refactors of the DDL handlers.

Cons:
- Event triggers are global per database; the function has to filter to tenant schemas (skip \`public.*\` migrations, skip platform-owned tables).
- Has to handle \`CREATE TABLE … LIKE\`, partitions, inherited tables, and other edge cases the typed handlers never see.
- Owner mismatch: the trigger runs as the calling role; needs SECURITY DEFINER to insert into \`schema_changes\` if the caller (e.g. \`eurobase_gateway\`) doesn't have INSERT on that table.

**B. Wrapper around the SQL runners** — partial fix.

\`query.HandlePlatformSQL\` / \`HandleSQL\` could parse the executed SQL post-hoc and call \`logSchemaChange\` for any DDL-shaped statement. Simpler to implement but:
- Re-implements DDL parsing in Go (fragile).
- Doesn't cover direct DB access.
- Doesn't cover SDK DDL paths that don't use \`HandleSQL\` (e.g. \`HandleDDL\` on \`/v1/db/schema/tables\`).

Recommend **A**. \`B\` is a stopgap if A is too much for a sprint.

## Notes

- The existing backfill (now NULL-stamped per #119) becomes mostly redundant once the event trigger is in place — newly-created tables get a real timestamp. The backfill stays useful for tables that existed *before* the event trigger was installed.
- The trigger has to be installed in every tenant DB. Since Eurobase is single-DB multi-schema, that's once globally. Migration \`000055_*.up.sql\` (or whatever slot is free) ships it.
- \`logSchemaChange\` from the typed handlers can stay as-is — it'd write duplicate rows for the same DDL (one from the handler, one from the event trigger). Either deduplicate (PRIMARY KEY constraint won't help; the rows have different ids) or have the typed handlers stop logging and let the trigger be the single source of truth. Probably the latter — single source is cleaner.

## Severity

Medium. Real impact for any tenant who uses the SQL editor: they make changes, Migration History claims nothing happened. Confusing for compliance / audit conversations. Low-risk to fix; high-trust gain.

### PR DESCRIPTION
**Pull request by STGime**
Option B from issue #120, after option A (Postgres event triggers) bounced off Scaleway managed Postgres's no-superuser policy (#121 → reverted in #122).

## Approach

Adds a SQL-text detector that scans every successful runner call for DDL shapes and writes one \`schema_changes\` row per detected operation. Wired into all three runner entry points so the gateway-mediated paths — **console SQL editor, MCP \`runSQL\`, SDK \`runSQL\`** — all flow into Migration History.

## What the detector recognises

- \`CREATE TABLE [IF NOT EXISTS] [schema.]name\`
- \`DROP TABLE [IF EXISTS] [schema.]name\`
- \`ALTER TABLE … ADD COLUMN [IF NOT EXISTS] name type\` → \`add_column\`
- \`ALTER TABLE … DROP COLUMN name\`
- \`ALTER TABLE … RENAME TO new_name\` → \`rename_table\`
- \`ALTER TABLE … RENAME COLUMN old TO new\` → \`alter_column\` (\`rename_to\` in detail)
- \`ALTER TABLE … ALTER COLUMN name {TYPE … | SET/DROP NOT NULL | SET/DROP DEFAULT}\`
- \`CREATE [UNIQUE] INDEX [name] ON table\`
- \`DROP INDEX name\` (target table unresolvable — same caveat as the event-trigger \`sql_drop\` path)

False-positive guards: single-quoted strings, line comments, block comments (nest-aware), and dollar-quoted blocks are stripped before matching. \`SELECT 'CREATE TABLE …'\` and \`CREATE FUNCTION\` bodies containing DDL text don't trigger audit rows.

## Wired into

- \`HandleSQL\` (\`/v1/db/sql\` — SDK). Read-only path skips detection because \`ValidateSelectOnly\` rejects DDL upstream; the write path (e.g. \`eb.db.sql()\` via secret key) audits.
- \`HandlePlatformSQL\` (\`/platform/projects/{id}/data/sql\` — console SQL editor + MCP \`runSQL\`).
- \`HandlePlatformSQLTransaction\` (\`/platform/projects/{id}/data/sql/transaction\` — per-statement loop).

## Documented in the console

New **Migration history** subsection in the Database docs page (Database → Migration History timeline). Lists what's tracked, the audited paths, and a yellow callout explaining the limitation that direct DB access (psql / external ORMs / Scaleway DB console) bypasses the audit — plus *why* (event-trigger superuser requirement) and which paths to prefer for a complete audit trail.

## Limitations vs option A

Direct DB access still bypasses the audit. The detector only runs when SQL flows through the gateway. Documented in the issue and the user-facing docs as acceptable given the managed-Postgres privilege ceiling.

## Test plan

- [x] \`go test ./...\` clean (21 table-driven cases for the detector + existing query/gateway suites)
- [x] \`npm run build\` (console) clean
- [ ] Merge + deploy
- [ ] On LiveStylist: \`ALTER TABLE app_users ADD COLUMN test_col text\` via MCP \`runSQL\` → entry appears in Migration History with the correct timestamp + \`type: text\` in detail
- [ ] On any project: \`DROP TABLE\` from the SQL editor → \`drop_table\` row appears
- [ ] Docs page renders the new subsection with the yellow callout

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/query/ddl_detect.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+				Action:    "drop_table",
+				TableName: unquoteIdent(m[1]),
+				Detail:    map[string]any{"source": "sql"},
+			}
+		},
+	},
+	{
+		// ALTER TABLE [IF EXISTS] [ONLY] [schema.]name ADD COLUMN [IF NOT EXISTS] name type
+		pattern: regexp.MustCompile(`(?is)\bALTER\s+TABLE\s+(?:IF\s+EXISTS\s+)?(?:ONLY\s+)?` + optSchema + `(` + ident + `)\s+ADD\s+COLUMN\s+(?:IF\s+NOT\s+EXISTS\s+)?(` + ident + `)\s+([A-Za-z][A-Za-z0-9_ ]*)`),
+		build: func(m []string) *DetectedDDL {
```
Comment (STGime) [inline internal/query/ddl_detect.go:114]: **Limitation (low, acceptable for v1):** Multi-subcommand ALTER TABLE won't detect the second clause. E.g.:

```sql
ALTER TABLE users ADD COLUMN a text, ADD COLUMN b int;
```

The regex requires `\bALTER\s+TABLE` at the start, so only the first `ADD COLUMN a text` matches — the `, ADD COLUMN b int` part lacks the `ALTER TABLE` prefix and is missed. Each regex uses `FindAllStringSubmatch`, but the second ADD COLUMN isn't a standalone match.

Not a blocker — multi-clause ALTER is uncommon in console/MCP usage, and the first column is still captured. Worth a comment or test case documenting the gap so a future contributor doesn't wonder why.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: internal/query/ddl_detect.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+	s := sql
+	i := 0
+	n := len(s)
+	for i < n {
+		c := s[i]
+		switch {
+		case c == '\'':
+			// Single-quoted string. Postgres allows '' as an escape.
+			j := i + 1
+			for j < n {
```
Comment (STGime) [inline internal/query/ddl_detect.go:262]: The stripper is thorough — nest-aware block comments, `''` escape handling in single-quoted strings, and dollar-quoted bodies all covered. The space-padding to preserve regex offsets is a nice touch that avoids the "two-pass with offset maps" complexity.

One edge case worth considering for a future test: **E-string syntax** `E'...'` or `e'...'` — Postgres supports C-style backslash escapes inside E-strings (`E'it\'s'`). The current scanner handles `''` but not `\'`. This is very rare in practice (especially from a SQL runner UI) so not a blocker.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: internal/query/sql_handler.go [hunk 3]
```diff
@@ -78,6 +108,13 @@ func HandlePlatformSQLTransaction(engine *QueryEngine) http.HandlerFunc {
...
 		}
 
+		// Closes #120 for the multi-statement path. Each statement
+		// committed successfully (the whole tx did), so audit any DDL
+		// shapes we can recognise.
+		for _, stmt := range req.Statements {
+			auditDDLFromSQL(r, engine, stmt)
+		}
+
 		jsonResponse(w, SQLTransactionResponse{
```
Comment (STGime) [inline internal/query/sql_handler.go:118]: Good placement — auditing after the successful transaction path means we only log DDL that actually committed. If the tx rolls back, the early-return error path is taken and we never reach here.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: console/src/routes/(app)/docs/+page.svelte [hunk 4]
```diff
@@ -411,6 +411,24 @@ const eb = createClient({'{'}
...
+						Migration History tracks DDL that reaches the platform via the gateway. Schema changes made through a <strong>direct database connection</strong> &mdash; for example, <code class="rounded bg-amber-100 px-1 py-0.5 text-[11px] font-mono">psql</code> connected to your <code class="rounded bg-amber-100 px-1 py-0.5 text-[11px] font-mono">DATABASE_URL</code>, an ORM/migration tool running on your own infrastructure, or the Scaleway database console &mdash; <strong>do not appear</strong> in the history. They are real changes against your schema and your code/queries should still work, but the audit trail won't show them.
+					</p>
+					<p class="mt-2 text-xs text-amber-800 leading-relaxed">
+						This is a deliberate trade-off: capturing every change uniformly would require Postgres superuser privileges, which managed-Postgres providers (Scaleway, RDS, etc.) reserve for themselves. If you need a complete audit trail, prefer the gateway-mediated paths (SQL Runner / SDK / MCP) over direct DB access for any schema-changing operation.
+					</p>
+				</div>
+
 				<h3 class="text-lg font-semibold text-gray-900 mt-6">Using the SDK</h3>
```
Comment (STGime) [inline console/src/routes/(app)/docs/+page.svelte:432]: Clear, honest documentation. The callout explaining *why* (managed Postgres won't grant superuser for event triggers) is exactly the right level of transparency — users understand the constraint isn't a design shortcut.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by STGime — COMMENTED** [review]
Pragmatic option-B after the event trigger approach bounced off Scaleway's no-superuser policy. The regex-based detector is well-engineered: thorough string/comment stripping (nest-aware block comments, dollar-quoting, `''` escapes), 21 table-driven test cases including false-positive guards, and clean wiring into all three SQL runner entry points.

4 inline comments:

1. **Multi-subcommand ALTER TABLE gap (low)** — `ALTER TABLE t ADD COLUMN a text, ADD COLUMN b int` only detects the first column. The regex requires `\bALTER\s+TABLE` as a prefix, so the comma-separated second clause is missed. Acceptable for v1 (uncommon in console/MCP usage), but worth a comment or test case documenting the gap.

2. **E-string syntax edge case (informational)** — `E'it\'s'` uses backslash escapes that the stripper doesn't handle (it only handles `''`). Very rare from a SQL runner UI, not a blocker.

3. **Transaction audit placement** — Correctly placed after the successful commit path so rolled-back DDL is never audited.

4. **Docs callout** — Honest, clear documentation of the limitation and the *why* behind it.

No blockers — the test coverage is solid, the false-positive guards work, and the `logSchemaChange` reuse means the audit rows are consistent with typed-handler DDL. LGTM.

---
_Generated by [Claude Code](https://claude.ai/code)_
